### PR TITLE
Mccalluc/workspace login

### LINKED ISSUE #2629: Try to authenticate against workspaces
From Juan:
> you should be able to start using the DEV instance of the REST API: https://workspaces-api.dev.hubmapconsortium.org/ ... I would ask that you try to authenticate using the “Globus User Token Authentication” method in the [Authorization Walkthrough](https://docs.google.com/document/d/1h88wz4-PhF5JyozFGMoqyamfR0xDhrUkaxeL4Fcu1Cc/edit#heading=h.vefyblc8gqxd) and let me know once you’ve done so, so that I can check the user that got created. I want to see how your particular user gets created since you mentioned that the email address reported by Globus doesn’t quite fit the design.

(So it's not so much about creating code against this as walking though the process and seeing what the results are.)

### PR DESCRIPTION
**Pull request by mccalluc**
- Fix #2629

Filing as draft. Wait to merge until:
- There's a production workspaces instance
- The login process is fully documented on that repo (https://github.com/hubmapconsortium/user_workspaces_server/issues/27 and https://github.com/hubmapconsortium/user_workspaces_server/issues/26)

I don't think it's necessary to wait until all functionality is available before merging this.

### CODE REVIEW

**Comment by mccalluc** [comment]
Nils encourages us to hit more of the endpoints: But don't merge until there is a production instance. Continue work on this PR, but continue to hold off on merge.

**Comment by mccalluc** [comment]
When test is run individually, it passes:
```
$ node_modules/jest/bin/jest.js --testMatch '**/Files.spec.js'
 PASS  app/static/js/components/detailPage/files/Files/Files.spec.js (6.157 s)
  ✓ handles DUA flow (3237 ms)
  ✓ does not display file browser when files prop is undefined (29 ms)
```
But as part of larger set of tests, it fails:
```
$ node_modules/jest/bin/jest.js --testMatch '**/detailPage/**/*.spec.js'
...
Summary of all failing tests
 FAIL  app/static/js/components/detailPage/files/Files/Files.spec.js (23.776 s)
  ● handles DUA flow

    : Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 5000 ms timeout specified by jest.setTimeout.Error:
```

debugging strategies:
- Add console.log to figure out where the test gets stalled.
- Identify a smaller set of tests which reproduce the failure: What's the addition that makes a difference?
- Click through DUA by hand and see if there are problems.
- Does the async on the test make a difference?
- Skim this PR and see if DUA code was changed inadvertantly.

**Comment by mccalluc** [comment]
Passing locally now: The problem was probably related to something I was doing in the Menu... but we don't want to expose any links to the workspaces quite yet, so I commented all of that out.

FYI: I started work on the machinery to create a workspace, but I haven't actually seen it work yet. Assuming this passes CI now, I'd like this to be reviewed sooner rather than later, but I'll try to see this afternoon whether workspace creation works.

File: context/app/static/js/components/workspaces/WorkspacesList.jsx [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        </FlexBottom>
+      </SeparatedFlexRow>
+      <Paper>
+        {`TODO: Use token "${workspacesToken}" with "${workspacesEndpoint}"`}
+        <hr />
+        {JSON.stringify(workspacesList)}
+      </Paper>
+    </>
+  );
+}
```
Comment (john-conroy) [inline context/app/static/js/components/workspaces/WorkspacesList.jsx:94]: I think `SectionButtonRow` in `shared-styles` achieves what you want here?

File: context/app/static/js/pages/Workspaces/Workspaces.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+import WorkspacesList from 'js/components/workspaces/WorkspacesList';
+
+function Workspaces() {
+  const { isAuthenticated } = useContext(AppContext);
+  return (
+    <>
+      <SectionHeader variant="h1" component="h1">
+        My Workspaces
+      </SectionHeader>
+      {/* eslint-disable-next-line no-undef */}
```
Comment (john-conroy) [inline context/app/static/js/pages/Workspaces/Workspaces.jsx:16]: Is this needed?

File: context/app/static/js/pages/Workspaces/Workspaces.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      {!isAuthenticated ? (
+        <Description padding="20px">
+          The workspaces feature is only available if logged in. <LightBlueLink href="/login">Log in</LightBlueLink> to
+          view saved workspaces or to begin a new workspace.
+        </Description>
```
Comment (john-conroy) [inline context/app/static/js/pages/Workspaces/Workspaces.jsx:21]: Filed https://github.com/hubmapconsortium/portal-ui/issues/2677

File: context/app/static/js/components/workspaces/WorkspacesList.jsx [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+import { SeparatedFlexRow, FlexBottom } from './style';
+import { createNotebookWorkspace } from './utils';
+
+// TODO: Copy-and-paste from SummaryData/style
+const StyledButton = styled(WhiteBackgroundIconButton)`
+  height: 36px;
+`;
+
+function useWorkspacesList() {
```
Comment (john-conroy) [inline context/app/static/js/components/workspaces/WorkspacesList.jsx:18]: We've used separate files for hooks and styles and I'd prefer to stay consistent until we've decided to move away from that.

File: context/app/static/js/components/workspaces/style.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+  display: flex;
+  justify-content: space-between;
+  margin-bottom: ${(props) => props.theme.spacing(0.5)}px;
+`;
+
+const FlexBottom = styled.div`
+  display: flex;
+  align-items: flex-end;
+`;
+
```
Comment (john-conroy) [inline context/app/static/js/components/workspaces/style.js:17]: Are some of these styles no longer needed with the addition of `SpacedSectionButtonRow`?

**Review by john-conroy — APPROVED** [review]

**Pull request merged**